### ISSUE TITLE
CastStateListener to be used from Fragment

### ISSUE THREAD

**Issue by harshmittal29** [googlecast/CastVideos-android#80]
I want to use CastStateListener in a fragment to check if casting devices are available or not.
Code used in Fragment is 

```
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mCastContext = CastContext.getSharedInstance(mContext)
        mCastStateListener = CastStateListener { newState ->
            if (newState != CastState.NO_DEVICES_AVAILABLE) {
                castDevicesAvailable = true
            }
        }
    }

    override fun onResume() {
        super.onResume()
        mCastContext?.addCastStateListener(mCastStateListener)
    }


    override fun onPause() {
        super.onPause()
        mCastContext?.removeCastStateListener(mCastStateListener)
    }
```

This code does not give me a call back inside CastListner when used in Fragment but it works fine when I use it in an Activity or Fragment.

I am using custom view 

<```
androidx.mediarouter.app.MediaRouteButton
                    android:id="@+id/media_route_button"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:layout_gravity="center"
                    android:actionButtonStyle="@android:style/Widget.Holo.Light.MediaRouteButton"
```

I want to hide/show the view based on cast devices available

**Comment by anjaneesh** [comment]
I don't think it is a good idea to use CastStateListener in a fragment. Because an activity can host multiple fragments. When a activity is paused, all fragments in it are paused, so are resumed. In the code, you add CastStateListener in fragment onResume and remove CastStateListener in fragment onPause. If the activity hosts multiple fragments, it is very easy to mess up the add/remove CastStateListener. So I think it is better to add/remove the CastStateListener to the activity life cycle

**Comment by harshmittal29** [comment]
I changed by initial implementation to have CastStateListener in application class.
In Fragments I do get a call back in my cast listener but only second time onResume is called.

My work flow state is as follows :
1. CastContext and CastStateListener are declared in onCreate.
2. in onResume I am adding CastListener and in onPause I am removing it.
3. When my fragment gets initialised, it goes to onResume. First time I do not get a callback to my CastStateListener. But when I pause my fragment and resumes it again, the CastStateListener is called.

My implementation is such that I only want to show cast button on my fragment, so I think it makes sense to call CastStateListener in fragment itself. Any thoughts?

**Comment by victor-munoz** [comment]
@harshmittal29 did you find a solution?
